apim-operator/pkg/controller/api/api_controller.go

```diff
@@ -2132,3 +2098,56 @@ func deleteCompletedJobs(namespace string) error {
 	}
 	return nil
 }
+
+//update configmaps with OwnerReference
+func updateConfMapWithOwner(r *ReconcileAPI, owner []metav1.OwnerReference, configMap *corev1.ConfigMap) error {
+	configMap.OwnerReferences = owner
+	errorUpdateinterceptConf := r.client.Update(context.TODO(), configMap)
+	if errorUpdateinterceptConf != nil {
+		return errorUpdateinterceptConf
+	}
+	return nil
+}
+
+//Hanldling interceptors to modify request and response flows
+func interceptorHandler(r *ReconcileAPI, instance *wso2v1alpha1.API, owner []metav1.OwnerReference,
+	jobVolumeMount []corev1.VolumeMount, jobVolume []corev1.Volume, userNameSpace string) (bool, []corev1.VolumeMount, []corev1.Volume, error) {
+
+	interceptorConfigmap, err := getConfigmap(r, instance.Spec.InterceptorConfName, userNameSpace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Interceptors are not defined
+			log.Info("interceptors are not defined")
+			return false, nil, nil, nil
+		} else {
+			// Error getting interceptors configmap.
+			log.Error(err, "error retrieving configmap "+instance.Name+"-interceptors")
+			return false, nil, nil, err
+		}
+	} else {
+		//mount interceptors configmap to the volume
+		log.Info("Mounting interceptors configmap to volume.")
+		jobVolumeMount = append(jobVolumeMount, corev1.VolumeMount{
+			Name:      interceptorsVolume,
+			MountPath: interceptorsVolumeLocation,
+			ReadOnly:  true,
+		})
+		jobVolume = append(jobVolume, corev1.Volume{
+			Name: interceptorsVolume,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: instance.Spec.InterceptorConfName,
+					},
+				},
+			},
+		})
+		//update configmap with owner reference
+		log.Info("updating interceptors configmap with owner reference")
+		errorUpdateinterceptConf := updateConfMapWithOwner(r, owner, interceptorConfigmap)
+		if errorUpdateinterceptConf != nil {
+			log.Error(errorUpdateinterceptConf, "error in updating interceptors config map with owner reference")
```

Comment:
Shall we split this to a new line? Seems like, the length is exceeded.

Reply:
Resolved